src/tqec/compile/detectors/compute.py

```diff
@@ -0,0 +1,260 @@
+            contained in the sub-template and cannot escape from it (which is
+            mostly equivalent to say that flows should not interact with qubits
+            on the border of the sub-templates).
+        database: existing database of detectors that is used to avoid computing
+            detectors if the database already contains them. If provided, this
+            function guarantees that the database will contain the provided
+            situation when returning (i.e., either it already contained the
+            situation or it has been updated **in-place** with the computed
+            detectors). Default to `None` which result in not using any kind of
+            database and unconditionally performing the detector computation.
+
+    Returns:
+        a collection of detectors that should be added at the end of the circuit
+        that would be obtained from the provided `template` and
+        `plaquettes_at_timestep`.
+    """
+    unique_subtemplates = template.get_spatially_distinct_subtemplates(
+        k, fixed_subtemplate_radius, avoid_zero_plaquettes=True
+    )
+    increments = template.get_increments()
+    # Each detector in detectors_by_subtemplate is using a coordinate system
+    # centered on the central plaquette origin.
+    detectors_by_subtemplate: dict[int, frozenset[Detector]] = {
+        i: compute_detectors_at_end_of_situation(
+            (subtemplate,)
+            if len(plaquettes_at_timestep) == 1
+            else (subtemplate, subtemplate),
+            plaquettes_at_timestep,
+            increments,
+            database,
+        )
+        for i, subtemplate in unique_subtemplates.subtemplates.items()
+    }
+
+    detectors_by_measurements: dict[frozenset[Measurement], Detector] = dict()
+    for i, row in enumerate(unique_subtemplates.subtemplate_indices):
+        for j, subtemplate_index in enumerate(row):
+            if subtemplate_index == 0:
+                continue
+            plaquette_origin = Position2D(j * increments.x, i * increments.y)
+            for d in detectors_by_subtemplate[subtemplate_index]:
+                offset_measurements = frozenset(
+                    m.offset_spatially_by(plaquette_origin.x, plaquette_origin.y)
+                    for m in d.measurements
+                )
+                # We use the convention of `cirq.GridQubit` where the first coordinate
+                # is the column and the second coordinate is the row. Note that the
+                # coordinates of detectors are derived from the qubits, which are
+                # already in the `GridQubit` convention. So we only need to swap the
+                # coordinates of the plaquette origin.
+                coordinates = (
+                    d.coordinates[0] + plaquette_origin.y,
+                    d.coordinates[1] + plaquette_origin.x,
+                    *d.coordinates[2:],
+                )
```

Comment:
With the current `GridQubit` I think we don't need exchange the `x`/`y` coordinates anymore? Need to be checked.

Reply:
I think we should agree on one standard, write it somewhere, and ensure that we use it everywhere. I will open an issue for that.

Reply:
I think we should agree on one standard, write it somewhere, and ensure that we use it everywhere. I will open an issue for that.
(sorry for the duplicate message)

Reply:
See #359